crates/transaction-pool/src/maintain/interfaces.rs

```diff
@@ -0,0 +1,253 @@
+
+    /// Mark the first event as received
+    fn on_first_event(&mut self);
+
+    /// Set dirty addresses to be reloaded
+    fn set_dirty_addresses(&mut self, addresses: HashSet<Address>);
+
+    /// Add addresses to the dirty set
+    fn add_dirty_addresses<I>(&mut self, addresses: I)
+    where
+        I: IntoIterator<Item = Address>;
+
+    /// Remove an address from the dirty set
+    fn remove_dirty_address(&mut self, address: &Address) -> bool;
+
+    /// Check if there are dirty addresses
+    fn has_dirty_addresses(&self) -> bool;
+
+    /// Start reloading addresses from state
+    fn start_reload_accounts<Spawner>(
+        &mut self,
+        client: Client,
+        at: BlockHash,
+        task_spawner: &Spawner,
+    ) where
+        Client: StateProviderFactory + Clone + 'static,
+        Spawner: reth_tasks::TaskSpawner + 'static;
+
+    /// Update metrics
+    fn update_metrics(&self);
+}
+
+impl<Client> DriftMonitoring<Client> for DriftMonitor
+where
+    Client: 'static,
+{
+    fn is_reloading(&self) -> bool {
+        Self::is_reloading(self)
+    }
+
+    fn state(&self) -> PoolDriftState {
+        Self::state(self)
+    }
+
+    fn set_state(&mut self, state: PoolDriftState) {
+        Self::set_state(self, state)
+    }
+
+    fn on_first_event(&mut self) {
+        Self::on_first_event(self)
+    }
+
+    fn set_dirty_addresses(&mut self, addresses: HashSet<Address>) {
+        Self::set_dirty_addresses(self, addresses)
+    }
+
+    fn add_dirty_addresses<I>(&mut self, addresses: I)
+    where
+        I: IntoIterator<Item = Address>,
+    {
+        Self::add_dirty_addresses(self, addresses)
+    }
+
+    fn remove_dirty_address(&mut self, address: &Address) -> bool {
+        Self::remove_dirty_address(self, address)
+    }
+
+    fn has_dirty_addresses(&self) -> bool {
+        Self::has_dirty_addresses(self)
+    }
+
+    fn start_reload_accounts<Spawner>(
+        &mut self,
+        client: Client,
+        at: BlockHash,
+        task_spawner: &Spawner,
+    ) where
+        Client: StateProviderFactory + Clone + 'static,
+        Spawner: reth_tasks::TaskSpawner + 'static,
+    {
+        Self::start_reload_accounts(self, client, at, task_spawner)
+    }
+
+    fn update_metrics(&self) {
+        Self::update_metrics(self)
+    }
+}
+
+/// Trait defining the interface for canonical chain event processing
+pub trait CanonProcessing<N, Client, P>
+where
+    N: NodePrimitives,
+    Client: StateProviderFactory + BlockReaderIdExt + ChainSpecProvider + Clone + 'static,
+    P: TransactionPoolExt<Transaction: PoolTransaction<Consensus = N::SignedTx>> + 'static,
+{
+    /// Process a reorg event - returns true if the event was a reorg and was processed
+    fn process_reorg<'a, M>(
+        &'a mut self,
+        event: CanonStateNotification<N>,
+        client: &'a Client,
+        pool: &'a P,
+        drift_monitor: &'a mut M,
+    ) -> Pin<Box<dyn Future<Output = bool> + Send + 'a>>
```

Comment:
can we use impl Future here?

Reply:
we store this future as part of `PoolMaintainer` state to manually poll it in its Future impl, I believe it could be possible to use impl here but could be problematic, wdyt?